client.go

```diff
@@ -306,6 +306,12 @@ func (c *Client) Create(cfg Function) (err error) {
 // Initialize creates a new Function project locally using the settings
 // provided on a Function object.
 func (c *Client) Initialize(cfg Function) (err error) {
+
+	// Create project root directory, if it doesn't already exist
+	if err = os.MkdirAll(cfg.Root, 0755); err != nil {
+		return
+	}
+
```

Comment:
I really like the idea of creating the directory in which a function will be deployed because it would ensure the function is in a clean fresh directory.  However, one small question:  what do we do when a user wants to start over, without also removing their git history (`.git`) or other ancillary config files (`.direnv`)?  This is why the code exists to check if a directory is _effectively_ empty:  contains no visible files, and no .faas.yaml, before allowing a re-run of `create` or `init`.